Python/bytecodes.c

```diff
@@ -790,11 +790,17 @@ dummy_func(
             // Deopt unless 0 <= sub < PyList_Size(list)
             DEOPT_IF(!_PyLong_IsNonNegativeCompact((PyLongObject *)sub));
             Py_ssize_t index = ((PyLongObject*)sub)->long_value.ob_digit[0];
+#ifdef Py_GIL_DISABLED
+            STAT_INC(BINARY_SUBSCR, hit);
```

Comment:
This should go after `DEOPT_IF(res_o = NULL);` to match the behavior of the default build.